src/Tribe/I18n.php

```diff
@@ -0,0 +1,217 @@
+	 * The main purpose of this method is to avoid a rat race against plugins and themes that will filter the locale
+	 * by attaching the filtering method or function at `PHP_INT_MAX`.
+	 *
+	 * @since TBD
+	 *
+	 * @param string   $locale The locale to set for the execution of the callback.
+	 * @param callable $do     The callable to execute in the context of a specific locale.
+	 * @param array    $args   A set of arguments that will be passed to the callback.
+	 *
+	 * @return mixed The callback return value, if any.
+	 */
+	protected function with_locale( $locale, callable $do, array $args = [] ) {
+		global $wp_filter;
+		$locale_filters_backup = isset( $wp_filter['locale'] ) ? $wp_filter['locale'] : [];
+		$wp_filter['locale']   = $locale_filters_backup instanceof \WP_Hook ? new \WP_Hook() : [];
+
+		$force_locale = static function () use ( $locale ) {
+			return $locale;
+		};
+
+		add_filter( 'locale', $force_locale );
+		$result = $do( ...$args );
+		remove_filter( 'locale', $force_locale );
+
+		// Restore the `locale` filtering functions.
+		$wp_filter['locale'] = $locale_filters_backup;
+
+		return $result;
+	}
+
+	/**
+	 * Compiles the translations for a set of strings iterating on a set of domains.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,array|string> $strings The set of strings to compile the translations for.
+	 * @param string|array<string>       $domains The domain(s) that should be used to compile the string translations.
+	 *
+	 * @return array<string|array> A map of the compiled string translations.
+	 */
+	 public function compile_translations( array $strings, $domains ) {
+		$cache_salts = [ $strings, $domains, get_locale() ];
+		$cache_key   = __METHOD__ . md5( serialize( $cache_salts ) );
+
+		$expiration_trigger = Cache_Listener::TRIGGER_UPDATED_OPTION;
+		$cached             = tribe_cache()->get( $cache_key, $expiration_trigger, false, DAY_IN_SECONDS );
+
+		if ( false !== $cached ) {
+			return $cached;
+		}
+
+		foreach ( (array) $domains as $domain => $file ) {
+			// Reload it with the correct language.
+			unload_textdomain( $domain );
+
+			if ( 'default' === $domain ) {
+				load_default_textdomain();
+			} else {
+				Common::instance()->load_text_domain( $domain, $file );
+			}
+
+			// Loop on the strings the build the possible translations.
+			foreach ( $strings as $key => $value ) {
+				$value = is_array( $value ) ? reset( $value ) : $value;
+				if ( ! is_string( $value ) ) {
+					continue;
+				}
+
+				// Make sure we have an array.
+				$strings[ $key ] = (array) $strings[ $key ];
+
+				// Grab the possible strings for default and any other domain.
+				if ( 'default' === $domain ) {
+					$strings[ $key ][] = __( $value );
+					$strings[ $key ][] = __( strtolower( $value ) );
+					$strings[ $key ][] = __( ucfirst( $value ) );
+				} else {
+					$strings[ $key ][] = __( $value, $domain );
+					$strings[ $key ][] = __( strtolower( $value ), $domain );
```

Comment:
* :no_entry_sign: The $text arg must be a single string literal, not "$value".
* :no_entry_sign: The $domain arg must be a single string literal, not "$domain".

<sub>hash: cf17f077c2b5ded9588de7ef7dd48d5d</sub>